feat: orderSeller-주문승인 기능 구현

axhtl opened the pull request:
주문승인(approveOrder) 기능을 코틀린으로 구현

- findByOrderId → findByIdOrNull 확장함수로 변경
- 서비스에서 if문을 사용하여 유효성 검사를 하던 코드를, 도메인의 메서드로 빼서 재사용성 높임
- 스웨거 어노테이션 추가

Ogu1208 committed:
feat: Category 기본 CRUD 구현

hjj4060 committed:
feat: 스프링 시큐리티 jwt 적용 WIP

hjj4060 committed:
feat: 스프링 시큐리티 jwt 적용 WIP

axhtl committed:
feat: orderSeller-주문승인 기능 구현

upswp commented on src/main/kotlin/org/store/clothstar/order/domain/Order.kt:
```diff
@@ -0,0 +1,38 @@
+    val paymentMethod: PaymentMethod,
+
+    @Embedded
+    var totalPrice: TotalPrice
```
컬럼에 JPA 컬럼에 부여하는 특징을 넣어보면 좋을 것 같습니다~!

upswp commented on src/main/kotlin/org/store/clothstar/order/domain/vo/TotalPrice.kt:
```diff
@@ -0,0 +1,21 @@
+@Embeddable
+data class TotalPrice(
+    @Column(name = "total_shipping_price")
+    var shipping: Int = 3000,
```
3000에 값의 의미를 주석으로 달아두면 좋을 것 같습니다

upswp commented on src/main/kotlin/org/store/clothstar/order/service/OrderSellerService.kt:
```diff
@@ -0,0 +1,26 @@
+
+    @Transactional
+    fun approveOrder(orderId: Long): MessageDTO {
+        val order = orderUserRepository.findByOrderId(orderId) ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)
```
ResponseStatusException을 쓴 이유가 궁금합니다

upswp commented on src/main/kotlin/org/store/clothstar/order/service/OrderSellerService.kt:
```diff
@@ -0,0 +1,26 @@
+
+        order.updateStatus(Status.APPROVE)
+        orderSellerRepository.save(order)
+        return MessageDTO(HttpStatus.OK.value(), "주문이 정상적으로 승인 되었습니다.")
```
MessageDTO를 return하는 위치는 Controller에서 처리가 되면 좋겠습니다.

upswp reviewed (changes requested):
고생하셨습니다! 이 부분 수정 부탁드립니다!

upswp commented on src/main/kotlin/org/store/clothstar/order/controller/OrderSellerController.kt:
```diff
@@ -0,0 +1,23 @@
+    private val orderSellerService: OrderSellerService
+) {
+
+    @Operation(summary = "(판매자) 주문 승인", description = "(판매자) 주문을 승인한다.")
```
판매자만 주문을 승인할 수 있으니까 인가에 대한 부분이 필요한 API로 생각듭니다

upswp replied:
커스텀 어노테이션을 만들어서 처리해보십시오!

upswp reviewed (changes requested):
인가에 대한 부분 수정 부탁드립니다!

hjj4060 committed:
feat: memberService 코틀린 적용 WIP

axhtl committed:
refactor: Optimize Imports

axhtl committed:
refactor: TotalPrice에 nullable=false 추가

axhtl committed:
refactor: TotalPrice - shipping 필드에 주석 추가

axhtl committed:
refactor: orderDetails의 타입을 MutableList로 변경

axhtl committed:
refactor: MessageDTO 위치 변경, OrderRepository 통일, findByOrderIdAndStatus 메서드 추가

- MessageDTO의 return 위치를 서비스에서 컨트롤러로 변경
- OrderSeller/UserRepository 삭제, OrderRepository로 통일
- findByOrderIdAndStatus 메서드 추가

hjj4060 committed:
feat: memberService 코틀린 적용 WIP

axhtl committed:
feat: 커스텀 예외처리 추가

axhtl committed:
refactor: Optimize Imports

hjj4060 committed:
feat: 이메일 적용

axhtl committed:
test: 판매자 주문승인 테스트 코드 작성

- 메서드 호출 테스트
- 예외처리 테스트) 주문번호가 존재하지 않을 때
- 예외처리 테스트) 주문이 '승인대기' 상태가 아닐 때

axhtl committed:
feat: 판매자 주문 취소 기능 구현

hjj4060 committed:
feat: address, seller 코틀린 전환

hjj4060 committed:
feat: 오류 수정

Ogu1208 committed:
feat: Product 도메인 엔티티 구현

Ogu1208 committed:
refactor: Category DTO, 엔티티 생성 refactor

Ogu1208 committed:
refactor: toEntity 로직 삭제

Ogu1208 committed:
test: CategoryService Test

Ogu1208 committed:
refactor: 중복 setter 수정

Ogu1208 committed:
refactor: id null

Ogu1208 committed:
fix: import 에러 해결 및 rebase

hjj4060 committed:
feat: develop rebase

hjj4060 committed:
Merge pull request #8 from ClothingStoreService/feature/spring-security-apply

Feature/spring security apply

Ogu1208 committed:
fix: product 도메인 매핑 오류 수정, 개행 수정

Ogu1208 committed:
fix: product 도메인 엔티티 DB 예약어 필드 네이밍 수정

Ogu1208 committed:
feat: product 도메인 레포지토리 생성

Ogu1208 committed:
refactor: 생성자 컨벤션 수정, 엔티티 연관관계 수정

axhtl committed:
style: 코드 리포맷팅

axhtl committed:
refactor: findByOrderId -> findByIdOrNull 변경, 주문상태가 WAITING이 아닐 경우 예외처리 방식 수정

- 예외처리 방식 변경: 서비스에서 if문으로 예외처리를 하지 않고, 도메인에서 validateForStatus 메서드를 만들어 예외처리

axhtl committed:
refactor: Optimize Imports, 코드 리포맷팅

axhtl committed:
refactor: 판매자 주문 승인/취소 테스트코드 오류 수정

axhtl committed:
refactor: Swagger 어노테이션 추가

axhtl committed:
refactor: 판매자 주문 승인/취소 유효성 메서드 변경: DeletedAt이 null이 아닌 주문 필터링

- validateForStatus -> validateStatusAndDeletedAt 변경

Ogu1208 committed:
refactor: product 도메인 연관관계 , 컨벤션 수정

Ogu1208 committed:
Merge branch 'feature/develop-before-merge' into feature/product-option-Item

Ogu1208 committed:
Merge pull request #5 from ClothingStoreService/feature/product-option-Item

feat: Product 도메인 - 엔티티 수정 설계 구현

hjj4060 committed:
feat: 오류 수정

axhtl committed:
refactor: Order 관련 예외처리 코드 - common 패키지로 이동

axhtl committed:
refactor: OrderSellerServiceTest 오류 수정

hjj4060 committed:
feat: product service WIP

axhtl committed:
refactor: validateForStatusWAITINGAndDeletedAt() 로직 수정

- 매개변수로 주문상태를 받지 않고, 주문상태가 WAITING이 아닌 경우에만 예외처리
